nautobot/tenancy/views.py

```diff
@@ -18,68 +19,64 @@
 #
 
 
-class TenantGroupListView(generic.ObjectListView):
-    queryset = TenantGroup.objects.all()
-    filterset = filters.TenantGroupFilterSet
-    filterset_form = forms.TenantGroupFilterForm
-    table = tables.TenantGroupTable
-
-
-class TenantGroupView(generic.ObjectView):
+class TenantGroupUIViewSet(NautobotUIViewSet):
+    bulk_update_form_class = forms.TenantGroupBulkEditForm
+    filterset_class = filters.TenantGroupFilterSet
+    filterset_form_class = forms.TenantGroupFilterForm
+    form_class = forms.TenantGroupForm
     queryset = TenantGroup.objects.all()
+    serializer_class = serializers.TenantGroupSerializer
+    table_class = tables.TenantGroupTable
+    object_detail_content = ObjectDetailContent(
+        panels=(
+            ObjectFieldsPanel(
+                section=SectionChoices.LEFT_HALF,
+                weight=100,
+                fields="__all__",
+            ),
+            ObjectsTablePanel(
+                weight=200,
+                section=SectionChoices.RIGHT_HALF,
+                table_class=tables.TenantTable,
+                table_filter="tenant_group",
+                exclude_columns=["tenant_group"],
+            ),
+        )
+    )
 
     def get_extra_context(self, request, instance):
         # Tenants
-        tenants = Tenant.objects.restrict(request.user, "view").filter(
-            tenant_group__in=instance.descendants(include_self=True)
-        )
-
-        tenant_table = tables.TenantTable(tenants)
-        tenant_table.columns.hide("tenant_group")
-
-        paginate = {
-            "paginator_class": EnhancedPaginator,
-            "per_page": get_paginate_count(request),
-        }
-        RequestConfig(request, paginate).configure(tenant_table)
+        context = super().get_extra_context(request, instance)
+        if self.action == "retrieve":
+            tenants = Tenant.objects.restrict(request.user, "view").filter(
```

Comment:
Is this chunk of code even needed still? I thought using `ObjectsTablePanel` could let it be autogenerated.

Is it just needed because we're having to use `instance.descendants` as the filter parameter? If so, a comment to that effect would be good.